JAVA/Decimal To Binary Using Stack/DecimalToBinary.java

```diff
@@ -0,0 +1,72 @@
+import java.util.Scanner;
+
+public class DecimalToBinary {
+
+    public static void main(String[] args) {
+        
+        Stack stk= new Stack();
+        Scanner scn = new Scanner(System.in);
+        
+        System.out.println("Wellcome to Programme - Decimal To Binary Using Stack \n\nEnter a Decimal Number : ");
+        int number = scn.nextInt(); // Getting an integer from user
+        
+        for(int x = number ; x != 0; x = x/2) {
+            stk.push(x%2);
+        }
+        System.out.print("Binary number equal to " + number + " : ");
+        while(!stk.isEmpty()) {
+            System.out.print(stk.pop().data);

```

Comment:
This not empty check should be with the stack DS itself.

Reply:
What do you mean?

Reply:
The pop() should also check if stack is empty or not.

This condition here is ok.